cms/admin/permissionadmin.py

```diff
@@ -38,6 +38,27 @@ class PagePermissionInlineAdmin(TabularInline):
     extra = 0  # edit page load time boost
     show_with_view_permissions = False
 
+    def has_change_permission(self, request, obj=None):
+        site = self.get_site(request)
+        if not obj:
+            return permissions.has_global_permission(request.user, site, 'can_change_pagepermission')
+        return page_permissions.user_can_change_page_permissions(request.user, page=obj, site=site)
```

Comment:
`obj` here is always `Page` model, not `PagePermission` seems like django works like that in `ModelAdmin` in `changeform_view>>_changeform_view>>_create_formsets>>get_formsets_with_inlines>>get_inline_instances(self, request, obj=None)`